packages/grpc-js/src/server-call.ts

```diff
@@ -455,7 +455,9 @@ export class Http2ServerCallStream<
     }
 
     if (err) {
-      err.metadata = metadata;
+      if (!err.hasOwnProperty('metadata')) {
+        err.metadata = metadata;
+      }
```

Comment:
Open questions:
1. should i comment this?
2. would it be more appropriate to merge the existing `.metadata` property and the `metadata` local?  Note that the native impl does *not* do this and only uses the `err.metadata` property alone: https://github.com/grpc/grpc-node/blob/fa20c5e565439a6131508fcce782fdd15946f394/packages/grpc-native-core/src/server.js#L62-L65

Reply:
1. A comment probably isn't necessary. I think it's clear what's happening here.

2.  The behavior here should just match what the native implementation does.